### DISCUSSION TITLE
Samsung BMS testing results and possible enhancements?

### DISCUSSION THREAD

**Discussion by jezza61** [ai-republic/bms-to-inverter#52] [General]
The Samsung BMS has been tested this morning on a new clean RPI 4B with dual CAN hat, and a fresh clean install.
The data read looks right now, and the data sent to inverter has all of the right numbers.
One question, what model BMS do we specify on the Inverter? there are several BMS models that send almost the same data, which one does the Samsung BMS is trying to emulate?  Or does it matter?
Second one, I would prefer to limit the max change voltage, to somewhere less than 90% SOC to avoid any potential issues with overcharing, and perhaps stop discharge at the voltage slightly higher than BMS sends as minimal level for to stop discharge. Is that at all possible, or should I look at the inverter for these limits?
We have lots of very hot days here, and more conservative settings could be better?  Perhaps others had a similar issue?

**Comment by ai-republic** [comment]
1. I believe you are using a SMA SI inverter. Then just choose the Samsung _SAMSUNG_CAN_ binding for your BMS and the _SMA_SI_CAN_ binding for your inverter. In the SI inverter I believe you can only choose Lithium batteries as option but no specific BMS which is ok since the application will send the BMS data to the inverter in the SMA protocol.
2. Best to try and set these in your inverter settings. They are not really a part of the application, but I could provide a plugin for that.

**Comment by jezza61** [comment]
Hi,
I am waiting for a DEYE inverter.The Samsung bms, has a master bms, which produces a combined number for all trays.I have all up 16 trays in the unit.The main unit 001, does all summaries. But because I have 80kwh of power I am terrified of overcharging it 😁Anyway thanks for your program. 
Best regards Jerry

**Comment by ai-republic** [comment]
I guess we answered this in the other thread. Good to hear its working well 😉
